### PR TITLE
Add stake-o-matic

### PR DESCRIPTION
**Pull request by mvines**
#### Problem
* [168 lines of bash](https://github.com/solana-labs/validator-tracker/blob/ff5ad652898e50b0e33dd42c2d9e6d8d0cd065fe/run.sh)

#### Summary of Changes
* 697 lines of Rust

----
jk.  This PR modernizes the staker that is being used by testnet and mainnet-beta, and unifies the workflow for both clusters.  Additionally it adds a baseline and bonus stake to the staker.   All current validators will now receive the baseline stake of 5,000 SOL, while only quality validators will receive the 50,000 SOL.   This should improve the situation on testnet significantly.  See the README.md in this PR for more details.

Conveniently stake-o-matic has been written with the assumption that two pending RPC features are resolved, this PR will remain a draft until that is the case.

TODO:
* [x] Add Discord notifier support, to de-mystify staking for TdS/mainnet-beta validators
* [ ] Need to add retry to RPC, https://github.com/solana-labs/solana/issues/9974.  Stake-o-matic sends a transaction once, and expects RPC to retry until the transaction blockhash expires.   However this currently is not the case so most testnet stake-o-matic transactions fail
* [x] Need a mechanism to dry-run a transaction again over RPC against the last rooted bank, https://github.com/solana-labs/solana/issues/8986.  Stake-o-matic doesn't bother to check if a stake account is in the right state to accept a delegation or deactivation -- instead it'll just dry-run the transaction, and if successful then actually submit it.
* [x] Rework the dashboard at http://bit.ly/solana-validators as the metrics datapoints emitted by this PR differ slightly from the bash implementation
* [ ] Restore the daily staking automation with the implementation in this PR

### CODE REVIEW

File: stake-o-matic/mainnet-beta-validator-whitelist.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+- forb5u56XgvzxiKfRt4FVNFQKJrd2LWAfNCsCqL6P7q
+- 8ebFZA8NPLBZD91CwsG1HWQsa2B5Ludgdyf5Hi3sYhhs
+- 4XspXDcJy3DWZsVdaXrt8pE1xhcLpXDKkhj9XyjmWWNy
+- ateamuvZX4iy2xYrNrMyGgwtTpWFiurNzHpmJMZwjar
+- 4tS3UZfuRHzXuPenvErtRPtnZfY1KHhT96JBCQsLzKqW
+- fishfishrD9BwrQQiAcG6YeYZVUYVJf3tb9QGQPMJqF
+- 3RXKQBRv7xKTQeNdLSPhCiD4QcUfxEQ12rtgUkMf5LnS
+- 2mMGsb5uy1Q4Dvezr8HK2E8SJoChcb2X7b61tJPaVHHd
+- E6cyDdEH8fiyCTusmWcZVhapAvvp2LK24zMLg4KrrAkt
+- 3fA6TU7fQCkNDDKYJeCY4Ag2gCatEsYnYL4SpkSDYfCw
```
Comment (danpaul000) [inline stake-o-matic/mainnet-beta-validator-whitelist.yml:62]:
```suggestion
- 3fA6TU7fQCkNDDKYJeCY4Ag2gCatEsYnYL4SpkSDYfCw
- 9xg5FUCSJvvFkUF2dLoSHAR72DnA3qQqzTzWhHibZ33c
- 9H61MfefWLuCr8aoUTrqLN94ZZnu68pEqMHhnYxFSB4y
- HavuVVDXXsJqMzPwQ4KcF5kFm2xqjbChhyi1bgGeCQif
- 8LSwP5qYbmuUfKLGwi8XaKJnai9HyZAJTnBovyWebRfd
- DDnAqxJVFo2GVTujibHt5cjevHMSE9bo8HJaydHoshdp
- 2Ue9zGmDnvYRrJNEjuAdNkbbickw6fKWtbeNM7T2rakg
- 2vj1Ggh29cQTCL8RGqKF3Mn1pUHd9GMUGz6VjPXfgaiH
- DNWsTLvsMixgUcbM93437U8DWmJ9bZikQeBxQLHbeH5L
- 5ya8UPujuXN8cys1EaaqMMH3auby3HTHZki73Q4Yfkff
- 2us4ysyNvYJHkYi7CtRuW413Mwi34kjjFQGhZDch4DEN
- 6iSLGrFY1zCktMVkALyGqcQVqp4rUmKkozHG23EXwPwt
- HzvGtvXFzMeJwNYcUu5pw8yyRxF2tLEvDSSFsAEBcBK2
- 9JJQN1WpJ8QvH6XK1xAMbSpgHSiwqBgWaeCh3ViEFmtN
- 6wsSvrZPbjWeNNZ92KWtj94pdHj8v8sRbKsu1ZSpztpP
- uEhHSnCXvWgtgvVaYscPHjG13G3peMmngQQ2ghC54i3
- G2TBEh2ahNGS9tGnuBNyDduNjyfUtGhMcssgRb8b6KfH
- 4ULWSuaNnhQntP3DVxg1xa4yeNLNpDnAw3gTtrhPHzEA
- Dq3piY2ZcBvNN84j2EhDLtTzRAw95za7Eau89pNcmSd5
- a1phaKk6UbG1P2ZCpfMVFUeRM5E2EZhGvUjqWHRsrip
```

File: stake-o-matic/testnet-validator-whitelist.yml [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+- B6ZramCQhQWcq4Vxo3H5y1MYvo37wRZiAwKk88qaYNiF
+- 5vxoRv2P12q4K4cWPCJkvPjg6jYnuCYxzF3juJZJiwba
+- 5rxRt2GVpSUFJTqQ5E4urqJCDbcBPakb46t6URyxQ5Za
+- 4Nh8T1d4YBZHEuQNRmFbLXPT5HbWicqPxGeKZ5SdAr4i
+- VzZqDXSDF8KbfAQUvizj52NTjRg9cfXCAZ1cUJt5614
+- DadnDZbFH5BHHRHD7TaobaSQ7QATXgvWegHUcZ7ZGzmW
+- 8yS3Zc45xptsaay9iaUSpfdb5gaKcQaKAShVvEUFKpeN
+- 6ZEbKFxTjEKGC9HUqzy9z4ccJ8Aq3ktPKEzHGDosQJo4
+- 8xFQP5mPt9Nzr5wMeUsH4azMM3zyGqrUn3LeopKSwptX
+- D52Q6Ap8RVMw1EvJYTdEABP6M5SPg98aToMcqw7KVLD9
```
Comment (danpaul000) [inline stake-o-matic/testnet-validator-whitelist.yml:128]:
```suggestion
- D52Q6Ap8RVMw1EvJYTdEABP6M5SPg98aToMcqw7KVLD9
- 4gEKnFpiJ8XC6DdFw4D65uYQeMF8x7KDqMrBPrDVjMPb
- 5FPQXMJxXKJmuShQCiTRYPzXL9LBfgphprhXR54pr2eu
- 55ofKaF1xdfgC9mB4zUhrffdx7CVoxTbNo7GeQLyj3YL
- AKqueA5Vfmf6BWTXuPdWxrYCDNPGi5gDLrNpdc1CSEzy
- DJvMQcb3ZtXC49LsaMvAo4x1rzCxjNfBfZtvkUeR4mAx
- G7NoACjYzRcrLibw3dbNoDCVoBR67ijm3tYeL5zL5qKY
- DsnqNtwKA817a2VQypWEzaRXY2soq5Jgc68MgFBMR35p
- FLjMwfY9ahmyc6VgQpokBrra5Ucf5KSK5YW5x9MMKnpS
- YYYYW8eKkmwQFhVGUKdBAnDQPuhMTpG7zwm9nikNndC
```

File: stake-o-matic/testnet-validator-whitelist.yml [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+- 4gEKnFpiJ8XC6DdFw4D65uYQeMF8x7KDqMrBPrDVjMPb
+- 5FPQXMJxXKJmuShQCiTRYPzXL9LBfgphprhXR54pr2eu
+- 55ofKaF1xdfgC9mB4zUhrffdx7CVoxTbNo7GeQLyj3YL
+- AKqueA5Vfmf6BWTXuPdWxrYCDNPGi5gDLrNpdc1CSEzy
+- DJvMQcb3ZtXC49LsaMvAo4x1rzCxjNfBfZtvkUeR4mAx
+- G7NoACjYzRcrLibw3dbNoDCVoBR67ijm3tYeL5zL5qKY
+- DsnqNtwKA817a2VQypWEzaRXY2soq5Jgc68MgFBMR35p
+- FLjMwfY9ahmyc6VgQpokBrra5Ucf5KSK5YW5x9MMKnpS
+- YYYYW8eKkmwQFhVGUKdBAnDQPuhMTpG7zwm9nikNndC
```
Comment (danpaul000) [inline stake-o-matic/testnet-validator-whitelist.yml:137]:
```suggestion
- YYYYW8eKkmwQFhVGUKdBAnDQPuhMTpG7zwm9nikNndC
- CPPVEbGFbX3XAThetvfveCE1vYLWUwwJGT7DxkPAWb8D
```

File: stake-o-matic/testnet-validator-whitelist.yml [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+- CPPVEbGFbX3XAThetvfveCE1vYLWUwwJGT7DxkPAWb8D
```
Comment (danpaul000) [inline stake-o-matic/testnet-validator-whitelist.yml:138]:
```suggestion
- CPPVEbGFbX3XAThetvfveCE1vYLWUwwJGT7DxkPAWb8D
- 8ZjS3d1bQihC3p5voM8by2xi5PoBNxzTJtaQ9rvxUbbB
```

File: stake-o-matic/testnet-validator-whitelist.yml [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+- HgmPwzNcY85HfrN3bYiqaypb6Nmf7ayaZEaivGY37913
+- HwFvyMbGLkiTUaT66cfL1FnJ26c9VqtpqAg4UbWSXtdq
+- URnkWZGiuB7jXbfCSuNSwir1qkn7sXjiSPeLPaXys7b
+- VzZqDXSDF8KbfAQUvizj52NTjRg9cfXCAZ1cUJt5614
+- YYYYW8eKkmwQFhVGUKdBAnDQPuhMTpG7zwm9nikNndC
+- aeypRbHck9k5zGuftrdSaUMfjJyTrjtaGJEtALt835y
+- fish2tcDFRiAM1cNKqF7b96WzQmmDRjsf5XD214MhpV
+- h4idQXKHx1SBgzXeYAsuDAFvLzyepV2yr1o9wxKsGdx
+- hQBS6cu8RHkXcCzE6N8mQxhgrtbNy4kivoRjTMzF2cA
+- xViooFwobBZE416moj2yFV6RJwGVaBDbCQJRfCyz2FS
```
Comment (danpaul000) [inline stake-o-matic/testnet-validator-whitelist.yml:139]:
```suggestion
- xViooFwobBZE416moj2yFV6RJwGVaBDbCQJRfCyz2FS
- 5H3sMCaSJdN2k1hyuFTzq2BrZHUq7CinTa82hJS6EDTf
```

File: stake-o-matic/testnet-validator-whitelist.yml [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+- 3ANJb42D3pkVtntgT6VtW2cD3icGVyoHi2NGwtXYHQAs
+- 3FhfNWGiqDsy4xAXiS74WUb5GLfK7FVnn6kxt3CYLgvr
+- 3W4fe5WTAS4iPzBhjGP8a1LHBTx8vbscqThXT1THqEGC
+- 3i7sS5McrJ7EzU8nbdA5rcXT9kNiSxLxhwyfuxbsDvBj
+- 3w6hQh7Ndx93eqbaEMLyR3BwqtRxT2XVumavvU93mcRk
+- 47UuTGPAQZX2HnVcfxKk8b1BtA4rRTduVaHnvxzQe6AJ
+- 4Bx5bzjmPrU1g74AHfYpTMXvspBt8GnvZVQW3ba9z4Af
+- 4GhLBaxr1oEHWpoGnWh3mcRXUkBU5EEQZv3L27c7ohoq
+- 4Nh8T1d4YBZHEuQNRmFbLXPT5HbWicqPxGeKZ5SdAr4i
+- 4Nh8T1d4YBZHEuQNRmFbLXPT5HbWicqPxGeKZ5SdAr4i
```
Comment (danpaul000) [inline stake-o-matic/testnet-validator-whitelist.yml:23]:
```suggestion
```

File: stake-o-matic/testnet-validator-whitelist.yml [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+- 5KZRD6hDCZd1dgM9rJLweFFRkHu3qAgBhHgQ96Wz1VSn
+- 5NH47Zk9NAzfbtqNpUtn8CQgNZeZE88aa2NRpfe7DyTD
+- 5NorYZBbtgbouD3toX3761ZGbaYTWrNSDNci4G4zV8eo
+- 5TkrtJfHoX85sti8xSVvfggVV9SDvhjYjiXe9PqMJVN9
+- 5rxRt2GVpSUFJTqQ5E4urqJCDbcBPakb46t6URyxQ5Za
+- 5ueaf3XmwPAqk92VvUvQfFvwY1XycV4ZFoznxffUz3Hh
+- 5vxoRv2P12q4K4cWPCJkvPjg6jYnuCYxzF3juJZJiwba
+- 6PwxMMGLFnAf9sjMHfVr15z9fntjYTNPxJ7gFhkFxkXi
+- 6ZEbKFxTjEKGC9HUqzy9z4ccJ8Aq3ktPKEzHGDosQJo4
+- 6ZEbKFxTjEKGC9HUqzy9z4ccJ8Aq3ktPKEzHGDosQJo4
```
Comment (danpaul000) [inline stake-o-matic/testnet-validator-whitelist.yml:43]:
```suggestion
```

File: stake-o-matic/testnet-validator-whitelist.yml [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+- D23NCAVxinE53BTemguZCheAqCdMGfNTUzWdoWvq4Xj
+- D2NjDkcv8Y1dWGdtWAKPT4em2D3sYzM8AzMTpCG1RVf7
+- D52Q6Ap8RVMw1EvJYTdEABP6M5SPg98aToMcqw7KVLD9
+- D71JRzjPpHipt8NAWnWb3yZoXezbkGXqSf7TVCir6wvT
+- DFb6qaAkd5DTnFVYLDjzJNfsUPygP8GHHebN1CBv25cf
+- DJvMQcb3ZtXC49LsaMvAo4x1rzCxjNfBfZtvkUeR4mAx
+- DKnZytVA5wKbNPYW1pvPpoE5YeSsxu12KJFa95gBAGm7
+- DaB3ZwVtGLzSjazk5STQEu3MkJR2nkK3tDdCPAvx9QpM
+- DadnDZbFH5BHHRHD7TaobaSQ7QATXgvWegHUcZ7ZGzmW
+- DadnDZbFH5BHHRHD7TaobaSQ7QATXgvWegHUcZ7ZGzmW
```
Comment (danpaul000) [inline stake-o-matic/testnet-validator-whitelist.yml:108]:
```suggestion
```

File: stake-o-matic/testnet-validator-whitelist.yml [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+- HgmPwzNcY85HfrN3bYiqaypb6Nmf7ayaZEaivGY37913
+- HwFvyMbGLkiTUaT66cfL1FnJ26c9VqtpqAg4UbWSXtdq
+- URnkWZGiuB7jXbfCSuNSwir1qkn7sXjiSPeLPaXys7b
+- VzZqDXSDF8KbfAQUvizj52NTjRg9cfXCAZ1cUJt5614
+- YYYYW8eKkmwQFhVGUKdBAnDQPuhMTpG7zwm9nikNndC
+- aeypRbHck9k5zGuftrdSaUMfjJyTrjtaGJEtALt835y
+- fish2tcDFRiAM1cNKqF7b96WzQmmDRjsf5XD214MhpV
+- h4idQXKHx1SBgzXeYAsuDAFvLzyepV2yr1o9wxKsGdx
+- hQBS6cu8RHkXcCzE6N8mQxhgrtbNy4kivoRjTMzF2cA
+- xViooFwobBZE416moj2yFV6RJwGVaBDbCQJRfCyz2FS
```
Comment (danpaul000) [inline stake-o-matic/testnet-validator-whitelist.yml:140]: This is the balance of keybase-based pubkeys that aren't whitelisted.  A bunch of the recent discord complaints of not getting whitelisted came from people who registered via the keybase form a month ago, but are just getting started now.  If it won't hurt the stake-o-matic, I'd like to pre-empt further complaints and add all of the addresses that have come in from keybase that aren't currently on the whitelist.

```suggestion
- xViooFwobBZE416moj2yFV6RJwGVaBDbCQJRfCyz2FS
- 8ASvX43WvDF4LHe2ioNUbn3A4ZRD9iC6rsxyjo5sVGLp
- k4wTUPNkKM6b3LDSh3KbXsMCj64LoGmMB4W4iw2shXE
- Csrv9JCbebTKu1uBWqkfwuPHwVCXsYDrQmeXf19onbsY
- EU7Kd1fhYgUP21nSkev4DfYJQc2PmFGRXBDXRL1pMyGK
- FRgKM5ZXu9e3U7RrPqKiksm4v7MWcy14EixGQKufd8oG
- DciwdVV1DXimdsgRGQuQ45zYVjZNaof6a6EZ1JjaCsvx
- 74U9LiSPv2gb8bBZSh5uVNf89W4wZ8zs9B8EvRVVwr87
- 59WHuha1QunWmupWhFA4vr3WMUe8BLN7dc8HUsJ4YC86
- FzMsVyZXVuMRCwirhnb12MS64JPdtVY5R3jCwQ6C5JpX
- 86hxQGfre1vvVsYP7jPBka8ySrECZQewjG3MfxoCoBYf
- 4YGgmwyqztpJeAi3pzHQ4Gf9cWrMHCjZaWeWoCK6zz6X
- 58J9ucd9Qc6gMD8QHh2sHTyJyD8kdjHRQZkEAyAZ72YA
- FQigEZ2cSQZPLNjK5NMbt7oquyg17Q27cqQBMEJr4dQf
- BV8sS1jn1AvGAptY5TxNZdcm7aa49MZCXSpXQjzjdnYG
- AmNRXJSSaGJrXaYBahD1PBoBR2ApkfLBAJ7SiRK4772R
- 9fMqL641B7nQZ1xktU35qXuFESxMB7pkqmqVANtovzmE
- 31cud34DHkL5zM4ZiHXgsrkAu13Jeck7ahvkPU9i4Jze
- 9waRyqWAoP68etU17DdWamgpTnPb3skY7Er9kRZMzCfS
- 2jYzt9Ly7dNzNpMV9sbHBNuwMEvVdSi9L8yJCTJT21ki
- 9mbQ9mrRjBGiUVnz9Tdf7PuAkWomw3GLZ6Sup3R93Gw8
- 7FnrBgjPb1y8PNjzRLihQWUvky37F7wkvRb7MUL89Q8P
- HS9Kg3kHiyNCgDLHLv62DThvqaqQUiptWkbW3PmTXxLk
- 5dLMRyPWx6rdPGZpZ7uuZiqry96dUT5yz48u62Gzugi6
- 2ZeChc7Res7fUVdcGCDTJfRd9N8R21hiBPLAuJsqHHwh
- AoUwfPuiEek2thVRDhMP7HbQb9rguyab4rDiz2NAfwwA
- 6t8zWy766tsHBVNxhMwsTGiEYkGtjaZncRU3vcSEYtHU
- 8FaFEcUFgvJns6RAU4dso3aTm2qfzZMt2xXtSgCh3kn9
- G9nU3uDUXkH5opUt4niAxv3qh5ojFAZLLfvQEMv4EFVJ
- 7yzVecfpWupdJwVQby3inMggGSotFSnSrhvPGPp6JGdU
- F48fyuAYvfJgY4eTXkxmJwwXP6jFVFmQkzt4N4TSiGGF
- 4vgoKb76Z2vj9V9z7hoQpZkkwJrkL1z35LWNd9EXSi2o
- 75yzn7njq32yY1ieCZxFNVFZQWtEbQHpaaG6dSZFfmX5
- 8XvoJswfYCxWf2WkUmNBjtDWwonmKugYhxBruNpSfS4a
- CcorN3BoG1XMZehZ9Xib9YLo4mcvo7pzeVurC28gYYqX
- CtxU5HwVbgspJVtWxwjuP8wXUMdkjYJ4EJwJ3jvZh4zu
- DV78gathrorcpWsWrUkWrWNowLXpizKsPBupStzeAJnL
- HvSBzPhtU8RqjmyqFTVZtM4seYHZisz11G2MR5ZDxzJN
- 5MGfrpdVsyifhn2x62j6PnBWQk2c5xTUK1o8KFrwPLWG
- 6TkKqq15wXjqEjNg9zqTKADwuVATR9dW3rkNnsYme1ea
- ES5G93qiEu5hCyMZMpFUhnbK4YZKhrZMN86Xfkub9YJL
- D5JqF3qkLkeJKKEi145oMseEGc1ym9cWKtBKtg4ZBBnN
- 5PLDu8auwqtMkHW9zdPsfUcvyESZ45umFc4r8cWUw3Zp
- Drkj3wbHHmE2iCnqXHKFTmwPkuSc4bsFdgAmqv6eXuWi
- 23SUe5fzmLws1M58AnGnvnUBRUKJmzCpnFQwv4M4b9Er
- 6zCt5z72rfN9sRk2hTgc1LeFDbEBfXYmW6xtSNmgyama
- HrKX44KVBJByNgaMBkgweKipFvQchGGkhixY6kNtKLDD
- BMZY98zbjg2ey4XNfhBQXhEuvVqzaJ1T3AKD2quL3wnK
- B5FNFrfrrfpBggFBp9h6Js93mrz1Tham1z1GgP3bDgNc
- 3pzTi41c6NAbZyTiyPEAQtmi2K5MyWZJMxx6nDvWPgnQ
- 4FZSiJpGgprsVxkzc2F8v3bgnRpk8Ez1Dq7ohXwY1q9V
- 2YLPihCDxqztR5be69jhoNDPMxV6KeTJ2X2LtVBXDgp4
- FiEEzxMnedqDnDRStwshHNQTrD9DykkCSmbMcreqQuCW
- 5uTcsQSrUffYo6RYSWj75SuGMkJ4v9x5RYuQoTc5aWGR
- 6nrkRvzUpTst8teZJawMFFHrmixJ2sxAUxPKrqoGwCB8
- 8uJR7fEaTiriYLe4YMFkeeADdRkexxm8jkFCGRjPBvJD
- CjhKCjNC1WUgBjAGst3D2XmSsWHvt3zGasasmogPTY6J
- Ag51oCkwGy5rkbGEYrcP9GDjvFGMJrEdLxvedLSTSR15
- HoMBSLMokd6BUVDT4iGw21Tnxvp2G49MApewzGJr4rfe
- 4WufhXsUhPc7cdHXYxxDrYZVVLKa9jCDGC4ccfmuBvu2
- 7yiFPLeyH2B2p3xGrr2Y5nmk8B52nEaa6j55Nk3u6648
- 3ckQZncmgmS1aZCC7Eot659hoBQ8k85ExddD7fu9qJ2c
- 2XAHomUvH3LFjYSxzSfcbwS73JgynpQHfapMNMJ8isL9
- 5qsT9h2TuTMPLtX6gcD2DG6mcZwechxmWhKRFHbCt6Pu
- 787PK2WaCUZCyYEmuYQSGmoxu7MyqK1usn43FfiVwhcB
- 3xUTkgPKNJZ3dkpDMV8zWV34BkmvKanguKipv6M9x2Mt
- 3wz211BhQAE2n5fjDQSStM2iSizhNRyJDNRkDEc1YwMF
- 5sjXEuFCerACmhdyhSmxGLD7TfvmXcg2XnPQP2o25kYT
- Co7UqfqzXzTjhBwvam3zhNi4p8dKtdSrfh6rQykoNMy7
- 368KipD4nBzVs4AizHj1iU4TErSSqmZaNGVyzHx8TVXM
- 9a95Qsywtw2LKK9GkiR9dpTnoPGMVrLBLTAMPnaUwDjY
- 8nvD1CUE48WdcmRdvbyWcM5LdJKRTNP3tXT6Qp2CSND5
- ACv5dTk7THbmUpHYGhgPzMhWr7oqHSkuPJpa5RfvmG5H
- DfTeDaxk4RufkVbykedVnqa1r9S3z3oKFYL3FFmPdr1o
- 3HitRjngqhAgVuNdFwtR1Lp5tQavbJri8MvKUq5Jpw1N
- 2CGskjnksG9YwAFMJkPDwsKx1iRAXJSzfpAxyoWzGj6M
- CM1c6z3pRNgHFcfZG4z3wE31jaR8c4gCYQBJVEoCUyq8
- 4gMboaRFTTxQ6iPoH3NmxLw6Ux3SEAGkQjfrBT1suDZd
- FNdoUuKVBigMFGpVvSMLXJB4FC7XQL1RjPUqUiwvPiCS
- FhPGvLZRVn97AsZ5grffU7qhaRXrmxD4WcJgbLYkJh1K
- 2tZoLFgcbeW8Howq8QMRnExvuwHFUeEnx9ZhHq2qX77E
- Dq5r3zG6XGBcXNDRSWPSc7DiWwjqcGoiVcEhZ9mXEAaV
- 4vXPjSaZfydRqhnM85uFqDWqYcFyA744R2tjZQN8Nff4
- 4ZtE2XX6oQThPpdjwKXVMphTTZctbWwYxmcCV6xR11RT
```

Comment (mvines) [reply stake-o-matic/testnet-validator-whitelist.yml:137]: The only harm is more baseline/bonus stake account created so need more SOL, and I only have enough TdS SOL for 24 more accounts right now

Comment (danpaul000) [reply stake-o-matic/testnet-validator-whitelist.yml:137]: Baseline stake is only for non-delinquent validators right?  So if some of these addresses are offline, then no harm.  If they are online and not whitelisted, we should work to get them staked along with everyone else, right?

Comment (mvines) [reply stake-o-matic/testnet-validator-whitelist.yml:137]: No som creates the baseline/bonus stake accounts upfront for everybody in the whitelist

**Comment by mvines** [comment]
This is working well enough to land and start getting it plumbed into buildkite for nightly automation.

The retry mechanism is still a work in progress.  It's currently implemented directly in the app instead of via #9974.   This is because I'm still seeing strange behaviour on testnet when retrying transactions and I want to debug this from app code instead of first building #9974 and having to debug from within RPC, which will be much slower.

@danpaul000 - note that I dropped the yaml whitelist files for a whitelist.rs file, as we don't have a mechanism to distribute yaml files in the binary release artifacts (and I want buildkite to consume the release, not have to build from source/etc)

**Pull request merged**